filters/fadein/fadein.go

```diff
@@ -0,0 +1,260 @@
+	}
+
+	var f fadeIn
+	switch v := args[0].(type) {
+	case int:
+		f.duration = time.Duration(v * int(time.Millisecond))
+	case float64:
+		f.duration = time.Duration(int(v) * int(time.Millisecond))
+	case string:
+		d, err := time.ParseDuration(v)
+		if err != nil {
+			return nil, err
+		}
+
+		f.duration = d
+	case time.Duration:
+		f.duration = v
+	default:
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	f.exponent = 1
+	if len(args) == 2 {
+		switch v := args[1].(type) {
+		case int:
+			f.exponent = float64(v)
+		case float64:
+			f.exponent = v
+		default:
+			return nil, filters.ErrInvalidFilterParameters
+		}
+	}
+
+	return f, nil
+}
+
+func (fadeIn) Request(filters.FilterContext)  {}
+func (fadeIn) Response(filters.FilterContext) {}
+
+// NewEndpointCreated creates a filter spec for the endpointCreated filter.
+func NewEndpointCreated() filters.Spec {
+	var ec endpointCreated
+	return ec
+}
+
+func (endpointCreated) Name() string { return EndpointCreatedName }
+
+func normalizeSchemeHost(s, h string) (string, string, error) {
+	// endpoint address cannot contain path, the rest is not case sensitive
+	s, h = strings.ToLower(s), strings.ToLower(h)
+
+	hh, p, err := net.SplitHostPort(h)
+	if err != nil {
+		// what is the actual right way of doing this, considering IPv6 addresses, too?
+		if !strings.Contains(err.Error(), "missing port") {
+			return "", "", err
+		}
+
+		p = ""
+	} else {
+		h = hh
+	}
+
+	switch {
+	case p == "" && s == "http":
+		p = "80"
+	case p == "" && s == "https":
+		p = "443"
+	}
+
+	h = net.JoinHostPort(h, p)
+	return s, h, nil
+}
+
+func normalizeEndpoint(e string) (string, string, error) {
+	u, err := url.Parse(e)
+	if err != nil {
+		return "", "", err
+	}
+
+	return normalizeSchemeHost(u.Scheme, u.Host)
+}
+
+func endpointKey(scheme, host string) string {
+	return fmt.Sprintf("%s://%s", scheme, host)
+}
+
+// endpointCreated("", <int,float64,string,time>)
+// endpointCreated("http://10.2.4.123:2156", $time)
+func (endpointCreated) CreateFilter(args []interface{}) (filters.Filter, error) {
+	if len(args) != 2 {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	e, ok := args[0].(string)
+	if !ok {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	s, h, err := normalizeEndpoint(e)
+	if err != nil {
+		return nil, err
+	}
+
+	ec := endpointCreated{which: endpointKey(s, h)}
+	switch v := args[1].(type) {
+	case int:
+		ec.when = time.Unix(int64(v), 0)
+	case float64:
+		ec.when = time.Unix(int64(v), 0)
+	case string:
+		t, err := time.Parse(time.RFC3339, v)
+		if err != nil {
+			return nil, err
+		}
+
+		ec.when = t
+	case time.Time:
+		ec.when = v
+	default:
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	// mitigate potential flakyness caused by clock skew. When the created time is in the future based on
+	// the local clock, we ignore it.
+	now := time.Now()
+	if ec.when.After(now) {
+		log.Errorf("Endpoint created time in the future: %v. Potential clock skew.", ec.when)
```

Comment:
Maybe add what will happen in the message to understand impact from the message itself.

Reply:
ok (the fade-in will happen, we just ignore the created time)